### PR TITLE
Fix empty handling in facts base

### PR DESCRIPTION
**Pull request by NilashishC**
Signed-off-by: NilashishC <nilashishchakraborty8@gmail.com>

- Fix how emptys are handled in facts base template.

### CODE REVIEW

File: roles/scaffold_rm_facts/templates/module_utils/network_os/facts/base.py.j2 [hunk 1]
```diff
@@ -111,7 +111,7 @@ class FactsBase(object):
                 child_val = self.generate_final_config(val)
                 if child_val:
                     dct = {key: child_val}
-            elif val is not None:
+            elif val not in [None, [], {}, (), '']:
```
Comment (trishnaguha) [inline roles/scaffold_rm_facts/templates/module_utils/network_os/facts/base.py.j2]: lgtm

**Comment by NilashishC** [comment]
> Can you update the tests directory with a freshly generated copy of the role?

Sure @cidrblock, on it.

**Comment by cidrblock** [comment]
A sample is at the bottom of the README too
 ```ansible-playbook -e rm_dest=./rmb_tests/roles/my_role \
                 -e structure=role \
                 -e model=models/myos/interfaces/myos_interfaces.yml \
                 site.yml```

File: roles/scaffold_rm_facts/templates/module_utils/network_os/facts/base.py.j2 (hunk 1 above)
Comment (cidrblock) [reply roles/scaffold_rm_facts/templates/module_utils/network_os/facts/base.py.j2]: lgtm!

**Pull request merged**